sklearn/decomposition/_sparse_pca.py

```diff
@@ -434,3 +439,8 @@ def fit(self, X, y=None):
         self.n_components_ = len(self.components_)
 
         return self
+
+    def _more_tags(self):
+        return {
+            "preserves_dtype": [np.float64, np.float32],
+        }
```

Comment:
Isn't the exact same definition of `_more_tags` inherited from `SparsePCA` just above?

Reply:
> Isn't the exact same definition of `_more_tags` inherited from `SparsePCA` just above?

Yes. it is same with  `SparsePCA`. But the origin is different.  `SparsePCA` and `MiniBatchSparsePCA` implement different `fit` methods. The `preserves_dtype` tag of `MiniBatchSparcsePCA` is not inherited from `SparsePCA`, but the tag stands a `MiniBatchSparcsePCA` specific property. So defining the tag independently would be fine.
(It would be better to make `BaseSparsePCA` class, then define `SpasePCA` and `MiniBatchSparsePCA` inherited from `BaseSparsePCA`)

Reply:
Yes. it is same with SparsePCA. But the origin is different. SparsePCA and MiniBatchSparsePCA implement different fit methods. The preserves_dtype tag of MiniBatchSparcsePCA is not inherited from SparsePCA, but the tag stands a MiniBatchSparcsePCA specific property. So defining the tag independently would be fine.
(It would be better to make BaseSparsePCA class, then define SpasePCA and MiniBatchSparsePCA inherited from BaseSparsePCA)

Reply:
So I think it is ok as is

Reply:
This is unneeded and can be removed.

```suggestion
```

Reply:
Ok!

Reply:
Removed